conversant/prompt_chatbot.py

```diff
@@ -186,11 +188,103 @@ def _update_max_context_examples(
             "tokens"
         )
 
+    def should_stop(self, response:str)-> bool:
+        """Given a response, decides if it ends with a stop sequence and should 
+        be terminated."""
+        for stop_seq in self.client_config["stop_sequences"]: #make into function
+            if response.endswith(stop_seq):
+                return True
+        return False
+    
+    def get_max_reruns(self):
+        """Returns the maximum number of times partial_reply should be invoked."""
+        return int(self.client_config["max_tokens"]/TOKENS_PER_REQUEST)
+    
+    def partial_reply(self, query: str, is_from_scratch: bool):
+        """Replies to a query given a chat history.
+
+        The reply is then generated directly from a call to a LLM.
+
+        Used for all responses besides the initial one. 
+
+        Args:
+            query (str): A query passed to the prompt chatbot.
+            
+            is_from_scratch (bool): Tells the chatbot if the reply
+            
+            should be generated from scratch or it is the continuation
+            
+            of the previous chunk
+            
+        Returns:
```

Comment:
Could you add that to the function itself? Something like:

`partial_reply(self, query: str, is_from_scratch: bool) -> Dict[TYPES], bool:`

Reply:
I think that gives an error for multiple return values 🤔

Reply:
Oh true, you would have to do something like
`-> Tuple[Dict[TYPES], bool]`

Reply:
yes, I just did, thank you! I am new to docstring, sorry for all the missing stuff, this won't be an issue next time!

Reply:
This is from [this extension](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) btw!